### PR TITLE
* add ssh action

### PR DESCRIPTION
**Pull request by hjanuschka**
adds an action to execute commands on remote host using ssh.

use-case: i am trying to port some of my deployment scripts, and there i needed to restart a webserver via ssh

samples:

```
ssh(
        host: "dev.januschka.com",
        username: "root",
                port: "322",
        commands: [
                     "ls && true",
                     "ls && false",
                     "ls && true"
                    ]
    )
```

executes all commands in the `commands` array, if one of the commands fails, it stops, and raises a exception.

returns a hash

``` ruby
{ 
:stdout  => "WHOLE STDOUT OF ALL COMMANDS",
:stderr => "WHOLE STDERR"
}
```

### CODE REVIEW

File: fastlane.gemspec [hunk 1]
```diff
@@ -31,6 +31,8 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'addressable', '~> 2.3.8' # Support for URI templates
   spec.add_dependency 'xcode-install', '~> 1.0.1' # Download new Xcode versions
 
+  spec.add_dependency 'net/ssh', '~> 3.0.2' # add SSH gem
```
Comment (KrauseFx) [inline fastlane.gemspec]: Please don't introduce new dependencies. Instead use the `Actions.verify_gem!('cocoapods')` method in your integration. Check out `cocoapods.rb` on how to do that :+1:

File: lib/fastlane/actions/ssh.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+            stderr << return_value[:stderr]
+            stdout << return_value[:stdout]
+          end
+        end
+        Helper.log.info "Succesfully executed #{params[:commands].count} commands on host: #{params[:host]}"
+        Helper.log.info "\n########### \n #{stdout} \n###############".magenta if params[:log]
+        Actions.lane_context[SharedValues::SSH_STDOUT_VALUE] = stdout
+        Actions.lane_context[SharedValues::SSH_STDERR_VALUE] = stderr
+        {stdout: Actions.lane_context[SharedValues::SSH_STDOUT_VALUE], stderr: Actions.lane_context[SharedValues::SSH_STDERR_VALUE]}
```
Comment (KrauseFx) [inline lib/fastlane/actions/ssh.rb]: Could you add a return to make it clear that you're returning this value

File: lib/fastlane/actions/ssh.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        # Define the shared values you are going to provide
+        # Example
+        [
+          ['SSH_STDOUT_VALUE', 'STDOUT of command'],
+          ['SSH_STDERR_VALUE', 'STDERR of command']
+        ]
+      end
+
+      def self.return_value
+        # If you method provides a return value, you can describe here what it does
```
Comment (KrauseFx) [inline lib/fastlane/actions/ssh.rb]: Please adds docs here

**Comment by KrauseFx** [comment]
Similar to the other PRs, please add Actions.md, make sure the tests are passing and `rubocop` verification works :+1:

**Comment by hjanuschka** [comment]
* rubocop is clean
 * tests pass
 * added entry to Actions.md
 * adopted feedback

File: docs/Actions.md [hunk 4]
```diff
@@ -1746,6 +1746,21 @@ It can process unit test results if formatted as junit report as shown in [xctes
...
+
+Lets you execute a series of commands on a remote host, if one of the commands in command-array returns non 0 - it fails.
+
+```ruby
+ssh(
+  host: "dev.januschka.com",
+  username: "root",
+  commands: [
+    "date && true",
+    "date && true"
```
Comment (KrauseFx) [inline docs/Actions.md]: Maybe provide 2 different examples? Why do you need the `&& true` here?

Comment (KrauseFx) [inline docs/Actions.md]: Oh because of the return status, right? But isn't date returning the right exit code?

File: lib/fastlane/actions/ssh.rb [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Allows remote command execution using ssh"
+      end
+
+      def self.details
+        # Optional:
```
Comment (KrauseFx) [inline lib/fastlane/actions/ssh.rb]: Please remove this comment

File: lib/fastlane/actions/ssh.rb [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+      def self.output
+        [
+          ['SSH_STDOUT_VALUE', 'Holds the standard-output of all commands'],
+          ['SSH_STDERR_VALUE', 'Holds the standard-error of all commands']
+        ]
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
```
Comment (KrauseFx) [inline lib/fastlane/actions/ssh.rb]: Please remove this comment

**Comment by KrauseFx** [comment]
Thanks for updating the PR, however we have a merge conflict here. Could you address the remaining comments before it's ready to be merged

**Comment by hjanuschka** [comment]
* merge conflict solved (had to put the documentation block at the end of ###misc)
* added 1  more sample - showing a failing ssh case
* removed usless comments

**Comment by KrauseFx** [comment]
This looks great, thanks again for all your work :+1:

**Pull request merged**